chopper_generator/lib/src/generator.dart

```diff
@@ -1,59 +1,54 @@
-///@nodoc
-import 'dart:async';
+import 'dart:async' show FutureOr;
 
 import 'package:analyzer/dart/constant/value.dart';
 import 'package:analyzer/dart/element/element.dart';
-import 'package:analyzer/dart/element/nullability_suffix.dart';
 import 'package:analyzer/dart/element/type.dart';
 import 'package:build/build.dart';
 import 'package:built_collection/built_collection.dart';
 import 'package:chopper/chopper.dart' as chopper;
+import 'package:chopper_generator/src/extensions.dart';
+import 'package:chopper_generator/src/utils.dart';
+import 'package:chopper_generator/src/vars.dart';
 import 'package:code_builder/code_builder.dart';
 import 'package:dart_style/dart_style.dart';
 import 'package:logging/logging.dart';
 import 'package:source_gen/source_gen.dart';
 
-const String _clientVar = 'client';
-const String _baseUrlVar = 'baseUrl';
-const String _parametersVar = r'$params';
-const String _headersVar = r'$headers';
-const String _requestVar = r'$request';
-const String _bodyVar = r'$body';
-const String _partsVar = r'$parts';
-const String _urlVar = r'$url';
+/// Code generator for [chopper.ChopperApi] annotated classes.
+class ChopperGenerator extends GeneratorForAnnotation<chopper.ChopperApi> {
+  const ChopperGenerator();
 
-final Logger _logger = Logger('Chopper Generator');
+  static final Logger _logger = Logger('Chopper Generator');
 
-class ChopperGenerator extends GeneratorForAnnotation<chopper.ChopperApi> {
   @override
   FutureOr<String> generateForAnnotatedElement(
     Element element,
     ConstantReader annotation,
     BuildStep buildStep,
   ) {
     if (element is! ClassElement) {
-      final String friendlyName = element.displayName;
       throw InvalidGenerationSourceError(
-        'Generator cannot target `$friendlyName`.',
-        todo: 'Remove the [ChopperApi] annotation from `$friendlyName`.',
+        'Generator cannot target `${element.displayName}`.',
+        todo:
+            'Remove the [ChopperApi] annotation from `${element.displayName}`.',
       );
     }
 
     return _buildChopperApiImplementationClass(annotation, element);
   }
 
-  bool _extendsChopperService(InterfaceType type) =>
+  static bool _extendsChopperService(InterfaceType type) =>
       _typeChecker(chopper.ChopperService).isExactlyType(type);
 
-  Field _buildDefinitionTypeMethod(String superType) => Field(
+  static Field _buildDefinitionTypeMethod(String superType) => Field(
         (method) => method
           ..annotations.add(refer('override'))
           ..name = 'definitionType'
```

Comment:
Isn't this also sort of a `Vars`?

Reply:
Yeah, these methods could be further extracted into their own utility classes.